spec.bs

```diff
@@ -784,57 +868,78 @@ Issue: This would ideally be replaced by a more descriptive algorithm in Infra.
-  directly part of. If that was a component auction, they're from the component auction. If that was
-  the top-level auction, then they're from the top-level auction. Component bidders do not get these
-  signals from top-level auctions since it is the auction seller joining the top-level auction,
-  instead of winning component bidders joining the top-level auction directly.
-1. TODO: [=map/Set=] |browserSignals|["dataVersion"] to contain the Data-Version from the trusted
-  bidding signals response headers if they were provided by the trusted bidding signals server
-  response and the version was consistent for all keys requested by this interest group.
-1. [=map/Set=] |browserSignals|["seller"] to |config|["{{AuctionAdConfig/seller}}"].
-1. TODO: [=map/Set=] |browserSignals|["topLevelSeller"] if there was a top-level seller.
-1. TODO: Run [=evaluating a reporting script=] with |buyerReportingScript| "reportWin", and «
-   |auctionSignals|, |perBuyerSignals|, |sellerSignals|, |browserSignals| ».
-1. TODO: Set URLs passed to sendReportTo() into the FencedFrameConfig so they can be fetched when
-  ad renders.
-
+<div algorithm>
+To <dfn>report win</dfn> given a [=leading bid info=] |leadingBidInfo|, a [=string=]
+|sellerSignals| and an [=ordered map=] |browserSignals| whose [=map/keys=] are [=strings=] and whose
+[=map/values=] are {{any}}:
+
+  1. Let |config| be |leadingBidInfo|'s [=leading bid info/auction config=].
+  1. Let |winner| be |leadingBidInfo|'s [=leading bid info/leading bid=].
+  1. Let |perBuyerSignals| be |config|'s [=auction config/per buyer signals=].
+  1. Let |buyer| be |winner|'s [=generated bid/interest group=]'s [=interest group/owner=].
+  1. Let |perBuyerSignalsForBuyer| be |perBuyerSignals|[|buyer|] if that member [=map/exists=],
+    and null otherwise.
+  1. [=map/Remove=] |browserSignals|["`desirability`"].
+  1. TODO: [=map/Set=] |browserSignals|["`interestGroupName`"] to |winner|'s [=interest group/name=]
+    if the tuple of interest group owner, name, bidding script URL and ad creative URL were jointly
+    k-anonymous.
+  1. TODO: [=map/Set=] |browserSignals|["madeHighestScoringOtherBid"] are based on the auction the interest group was
+    directly part of. If that was a component auction, they're from the component auction. If that was
+    the top-level auction, then they're from the top-level auction. Component bidders do not get these
+    signals from top-level auctions since it is the auction seller joining the top-level auction,
+    instead of winning component bidders joining the top-level auction directly.
+  1. [=map/Set=] |browserSignals|["dataVersion"] to |leadingBidInfo|'s
+    [=leading bid info/bidding data version=].
+  1. [=map/Set=] |browserSignals|["adCost"] to |winner|'s [=generated bid/ad cost=].
+  1. [=map/Set=] |browserSignals|["seller"] to |config|'s [=auction config/seller=].
+  1. [=map/Set=] |browserSignals|["topLevelSeller"] to |leadingBidInfo|'s

```

Comment:
topLevelSeller is only set in component auctions.

Reply:
|leadingBidInfo|'s [=leading bid info/top level seller=] is only set for component auction, otherwise it's null.
https://github.com/WICG/turtledove/pull/532/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaL589

https://github.com/WICG/turtledove/pull/532/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaR1646

But I can add a check to only set topLevelSeller when |leadingBidInfo|'s [=leading bid info/top level seller=] is not null.